packages/contracts/test/voting/multisig.ts

```diff
@@ -338,140 +456,189 @@ describe('Multisig', function () {
+
+        expect(await multisig.canExecute(id)).to.equal(false);
+
+        // `tryExecution` is turned on but the vote is not decided yet
+        let tx = await multisig.connect(signers[1]).approve(id, true);
+        expect(await findEvent(tx, DAO_EVENTS.EXECUTED)).to.be.undefined;
+
+        expect(await multisig.canExecute(id)).to.equal(false);
+
+        // `tryExecution` is turned off and the vote is decided
+        tx = await multisig.connect(signers[2]).approve(id, false);
+        expect(await findEvent(tx, DAO_EVENTS.EXECUTED)).to.be.undefined;
+
+        // `tryEarlyExecution` is turned on and the vote is decided
+        tx = await multisig.connect(signers[3]).approve(id, true);
+        {
+          const event = await findEvent(tx, DAO_EVENTS.EXECUTED);
+
+          expect(event.args.actor).to.equal(multisig.address);
+          expect(event.args.callId).to.equal(id);
+          expect(event.args.actions.length).to.equal(1);
+          expect(event.args.actions[0].to).to.equal(dummyActions[0].to);
+          expect(event.args.actions[0].value).to.equal(dummyActions[0].value);
+          expect(event.args.actions[0].data).to.equal(dummyActions[0].data);
+          expect(event.args.execResults).to.deep.equal(['0x']);
+
+          const prop = await multisig.getProposal(id);
+          expect(prop.executed).to.equal(true);
+        }
+
+        // check for the `ProposalExecuted` event in the voting contract
+        {
+          const event = await findEvent(tx, PROPOSAL_EVENTS.PROPOSAL_EXECUTED);
+          expect(event.args.proposalId).to.equal(id);
+        }
+
+        // calling execute again should fail
+        await expect(multisig.execute(id))
+          .to.be.revertedWithCustomError(multisig, 'ProposalExecutionForbidden')
+          .withArgs(id);
       });
-      await expect(
-        multisig.updateMultisigSettings({onlyListed, minApprovals: 0})
-      ).to.be.revertedWith(customError('MinApprovalsOutOfBounds', 1, 0));
-    });
 
-    it('should emit MultisigSettingsUpdated when updateMutlsigSettings gets called', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
+      it('emits the `ProposalExecuted` and `Executed` events', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, false);
+
+        await expect(multisig.connect(signers[3]).execute(id))
+          .to.emit(dao, DAO_EVENTS.EXECUTED)
+          .to.emit(multisig, PROPOSAL_EVENTS.PROPOSAL_EXECUTED)
+          .to.not.emit(multisig, MULTISIG_EVENTS.APPROVED);
+      });
+
+      it('emits the `Approved`, `ProposalExecuted`, and `Executed` events if execute is called inside the `approve` method', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
```

Comment:
It seems like we are changing code beyond the assertion library itself here?

The logic tested at the end seems unrelated

Reply:
This was because this PR branched off of #176. Now, that is merged, these changes are not present on this branch anymore.